dulwich/client.py

```diff
@@ -679,7 +679,29 @@ def send_pack(self, path, determine_wants, generate_pack_contents,
         :raises UpdateRefsError: if the server supports report-status
                                  and rejects ref updates
         """
-        raise NotImplementedError(self.send_pack)
+        from dulwich.repo import Repo
+        from dulwich.protocol import ZERO_SHA
+        
+        target = Repo(path)
+        old_refs = target.get_refs()
+        if not old_refs:
+            old_refs["capabilities^{}"] = ZERO_SHA
```

Comment:
I basically copied the implementation of [HttpGitClient.send_pack](https://github.com/jelmer/dulwich/blob/master/dulwich/client.py#L1007), the remote response containing this special key directly is passed to `determine_wants` next line, which is a callback passed in from outside, existing code from hg-git already depends on this key exists.

So removing the key need other types of clients do the same, and hg-git need to patch `determine_wants`.